### PR TITLE
ISSUE #4835 Support for PolyLines and Angle Measurements

### LINKED ISSUE #4835: Measurement - Angles and Polylines
### Description

This is for the frontend changes for https://github.com/3drepo/3D-Repo-Product-Team/issues/394 - support for PolyLine and Angle measurements.

### Goals

- [ ] As a user, I should be able to snap against two lines and create an angular measurement between them (regardless of angle)
- [ ] As a user, I should be able to draw multiple points as a polyline (without bends) and determine the straight distance along the entire line.
- [ ] Angle measurements should show up in the measurements list

### Tasks
- [x] update icons
- [x] Create polyline listitem
- [x] Wire up polyline to unity
- [x] Add polyline to measurements
- [x] Add polyline to measurements in issues
- [x] Add polyline to measurements in risks
- [x] Create angle listitem
- [x] Wire up angle to unity
- [x] Add angle to measurements
- [x] Add angle to measurements in issues
- [x] Add angle to measurements in risks

### Related Resources

Product Team Ticket: https://github.com/3drepo/3D-Repo-Product-Team/issues/394
Unity Ticket: https://github.com/3drepo/3drepounity/issues/516

[Angle Mockup](https://www.figma.com/file/tmkgqJQlhjV8oUdhQwkHzr/Measurements?type=design&node-id=30-102&mode=design&t=Mt3AX4q4kL6rKi5j-0)
[Polyline Mockup](https://www.figma.com/file/tmkgqJQlhjV8oUdhQwkHzr/Measurements?type=design&node-id=30-139&mode=design&t=wZiFEV7SU6Q0XIpx-0)

### PR DESCRIPTION
**Pull request by sebjf**
This fixes #4835

#### Description
This PR for the moment is just to test the Unity changes, though #4835 will contain frontend changes before its done too.

Both the PolyLine and Angle tools are available, and ready to hook-up via the `UnityUtil.setMeasureToolMode()` method.

```
UnityUtil.setMeasureToolMode("PolyLine")
UnityUtil.setMeasureToolMode("Angle")
```

The PolyLine creates a measurement entry correctly, as it uses the existing Distance Gizmo. The Angle mode creates a measurement alert (see below) but it has a new type, and so won't appear in the list yet.

```
{
	"uuid": "2fb27167-1fd5-40aa-bde2-a2de0b55944d",
	"positions": [
		[
			-1.0,
			1.0,
			-1.0
		],
		[
			-1.0,
			1.0,
			1.0
		],
		[
			1.0,
			1.0,
			1.0
		]
	],
	"value": 1.57079625,
	"color": [
		0.04705,
		0.18431,
		0.32941,
		1.0
	],
	"type": 2
}
```

#### Testing

The best way to test it before the buttons are added to the frontend is to start the Measure Tool as normal, and make a measurement with, e.g. the Point to Point mode. Then, using the console give one of the commands above to put the tool into the chosen mode.

### CODE REVIEW

**Comment by AsiteZane** [comment]
Thank you @sebjf - i can confirm that i've tested them both and they are both as expected.
The behaviour on how to create them was as expected, thank you :)

Thank you for your efforts on this!

File: frontend/assets/icons/measurements/polyline.svg.tsx
Comment (The-Daniel) [inline frontend/assets/icons/measurements/polyline.svg.tsx:1]: Polyline is not reset when changing model. 
In this image I have created a polyline on one model
image
After using the breadcrumbs to change to another model and starting a new polyline the previous measurement lines appear on the model (or rather floating around the model)
image

Comment (sebjf) [reply frontend/assets/icons/measurements/polyline.svg.tsx:1]: @The-Daniel can you confirm that this is produced by starting a polyline measurement, then changing using the breadcrumb *before* finishing the measurement? I can reproduce that, but not a completed measurement sticking around.

Comment (sebjf) [reply frontend/assets/icons/measurements/polyline.svg.tsx:1]: Or, do you have a video of how the bug is produced?

Comment (The-Daniel) [reply frontend/assets/icons/measurements/polyline.svg.tsx:1]: Yes, this is when changing before finishing the measurement. I couldn't actually figure out how to complete a polyline measurement. Is it possible?

Note: there is a similar bug with an existing polyline appearing when you disable the polyline before completion and then try to create a new one

https://github.com/3drepo/3drepo.io/assets/62481250/ccc4600c-9cdb-49da-b727-1e9093957f38

Comment (sebjf) [reply frontend/assets/icons/measurements/polyline.svg.tsx:1]: @The-Daniel I have pushed an updated viewer that should have fixed that bug, let me know if that resolves the issue you found!

Comment (sebjf) [reply frontend/assets/icons/measurements/polyline.svg.tsx:1]: @The-Daniel polylines are terminated by double clicking. The fix above should solve the turning the tool on and off issue too!

Comment (The-Daniel) [reply frontend/assets/icons/measurements/polyline.svg.tsx:1]: Yes, all seems fixed now!

**Review by The-Daniel — APPROVED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Pull request merged**